### ISSUE TITLE
Uncaught Error in callback mode leading to deadlock

### ISSUE THREAD

**Issue by simeonborko** [rogierschouten/async-lock#51]
Consider the following code:

```js
const AsyncLock = require('async-lock');
const lock = new AsyncLock;

lock.acquire("key", function (done) {
   throw new Error("error");
});
```

I would expect that `acquire()` would return rejected Promise but I got `Uncaught Error: error`, and the lock is not released.

The snippet is just slightly different from the one in the README which returns rejected Promise.
```js
lock.acquire(key, function() {
  throw new Error('error');
})
```

From your point of view, is my code just improper usage of the library or can we consider it as a bug? If so, I'd be happy to try to prepare PR.

**Comment by rogierschouten** [comment]
Looks like a bug to me, thank you for logging it and preparing PR. I cannot merge it this week but I will strive for next week

**Comment by rogierschouten** [comment]
Closed through #52
